Add script to download and install Tanzu Framework CLI

sriramandev opened the pull request:
Currently there is a lengthy way to install tanzu cli in the documentation. Provide an easier option to install tanzu cli by just calling a script which downloads the latest stable release and install the same and update the documentation to reflect the new way of installation.

### What this PR does / why we need it
This PR provides for a script which fetches the latest Tanzu Framework release CLI and installs the CLI binary and the plugins locally. Currently the script handles only MacOS and Linux flavours for AMD64 architecture. The documentation update reflects the new option to download and install the CLI.

### Which issue(s) this PR fixes

Fixes # [1640](https://github.com/vmware-tanzu/tanzu-framework/issues/1640)

### Describe testing done for PR

- On MacOS system ran the script to do a fresh installation
- On MacOS with already installed cli, made sure the script installs the latest one after deleting the cache and deleting the older version
- Followed the above two steps on Ubuntu 20.04

### Release note
<!--
     Please add a short text (limit to 1 to 2 sentences if possible) in the release-note block below if
     there is anything in this PR that is worthy of mention in the next release.

     See https://github.com/vmware-tanzu/tanzu-framework/blob/main/docs/release/release-notes.md#does-my-pull-request-need-a-release-note
     for more details.
-->
```release-note
Added a script to fetch and install the latest Tanzu Framework CLI binary for MacOS and Linux distributions. Updated the documentation to reflect the new approach to install the CLI.
```

### PR Checklist

<!-- Please acknowledge by checking that they are being followed -->

- [ ] Squash the commits into one or a small number of logical commits
      <!--
      This repository adopts a linear git history model where no merge commits are necessary. To
      keep the commit history tidy, it is recommended that authors be responsible for the decision
      whether to squash the PR's changes into a single commit (and tidy up the commit message in the
      process) or organizing them into a small number of self-contained and meaningful ones.
      -->
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access


### Additional information

#### Special notes for your reviewer

The script provided downloads the latest release from Github release page (using the Github API). Once downloaded, it calls then the existing install script in the hack directory to install the CLI and plugins.

Commit:
feat: 🎸 Add script to download and install tanzu cli

Currently there is a lengthy way to install tanzu cli in the documentation. Provide an easier option to install tanzu cli by just calling a script which downloads the latest stable release and install the same and update the documentation to reflect the new way of installation.

stmcginnis commented on .gitignore:
```diff
@@ -46,4 +46,6 @@ providers.tar.gz
\ No newline at end of file
+/packages/**/.imgpkg
+
+.idea
```
Please see the top of this file. ;)

sriramandev replied:
Have removed the changes. :)

stmcginnis commented on docs/cli/getting-started.md:
```diff
@@ -23,6 +23,7 @@ With [v0.14.0](https://github.com/vmware-tanzu/tanzu-framework/releases/tag/v0.1
 The API driven plugin discovery feature is enabled as default method to install the plugins. So, most users will just need to install the Tanzu CLI binary. The plugins can be installed and configured with `tanzu plugin sync` command. Learn more about this feature in the [design docs](../design/context-aware-plugin-discovery-design.md).
 
 ##### macOS/Linux
+###### Option 1: Manual download CLI binary from Github Releases
```
nit: Please use proper casing for GitHub.

sriramandev replied:
Changed the casing of GitHub as suggested.

stmcginnis commented on docs/cli/getting-started.md:
```diff
@@ -23,6 +23,7 @@ With [v0.14.0](https://github.com/vmware-tanzu/tanzu-framework/releases/tag/v0.1
 The API driven plugin discovery feature is enabled as default method to install the plugins. So, most users will just need to install the Tanzu CLI binary. The plugins can be installed and configured with `tanzu plugin sync` command. Learn more about this feature in the [design docs](../design/context-aware-plugin-discovery-design.md).
 
 ##### macOS/Linux
```
The markdown linter isn't going to like this two headers together like this. I don't think content is required, but it will at least need a blank line between them.

sriramandev replied:
Have added blank lines between the two headers. Running the linter now seems to be green. Please let me know if I have misinterpreted your comment here.

stmcginnis commented on docs/cli/getting-started.md:
````diff
@@ -67,7 +68,17 @@ The API driven plugin discovery feature is enabled as default method to install
   ```sh
   tanzu plugin list
   ```
+###### Option 2: Using install script
````
Same - need a blank line after the title line to make the linter happy.

sriramandev replied:
Added the blank lines and checked with doc-linter. Seems green

stmcginnis commented on hack/fetch-install-tf.sh:
```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+# Copyright 2021 VMware, Inc. All Rights Reserved.
```
s/2021/2022/

stmcginnis reviewed (commented):
Looks great! Pointed out some minor things, but overall looks really good. I need to test on a couple different environments yet, but looks like this is in good shape once the linting issues are addressed.

Commit:
refactor: 💡 Resolve comments from PR 1705

Resolve the comments in .gitignore, documentations and copyright related
changes suggested in PR 1705

BREAKING CHANGE: 🧨 N/A

navidshaikh reviewed (commented):
Looks good! Tested fresh install on macOS.

The script output/logs are verbose, we can suppress the responses returned by github APIs.

sriramandev committed:
Correct the usage of fetch and install cli script

The usage guideline in the comments are erroneous. Correct it to point to right usage

Co-authored-by: Navid Shaikh <shaikhnavid14@gmail.com>
